Throttler.pm

```diff
@@ -117,6 +117,23 @@ sub unlock {
 }
 
 ###########################################
+sub current_value {
+###########################################
+    my($self, %options) = @_;
+
+    if(exists $options{key}) {
```

Comment:
Almost there! These four lines checking for the key seem to have crept in via copy-and-paste from a different function, but you don't need them in current_value().